app/locations/trailing-history.js

```diff
@@ -0,0 +1,13 @@
+import HistoryLocation from '@ember/routing/history-location';
+
+export default class TrailingHistoryLocation extends HistoryLocation {
+  formatURL() {
+    let url = super.formatURL(...arguments);
+
+    if (url.includes('#')) {
+      return url.replace(/([^/])#(.*)/, '$1/#$2');
+    } else {
+      return url.replace(/\/?$/, '/');
+    }
```

Comment:
Let's make it clear to others that there's an early exit.

```suggestion
    if (url.includes('#')) {
      return url.replace(/([^/])#(.*)/, '$1/#$2');
    }

    return url.replace(/\/?$/, '/');
```

Reply:
but there isn't an "early exit" 🤔  both paths do different things but they are both doing something to the URL so I would be uncomfortable making this look like an early exit